### PR TITLE
Gem ransack for admin

### PR DESCRIPTION
**Pull request by nhat15**
## Related Tickets
- [#53513](https://edu-redmine.sun-asterisk.vn/issues/53513)

## WHAT (optional)
Add gem ransack for admin movie and genre page

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evid
Screenshot from 2022-09-19 14-54-07
ence (Screenshot or Video)
Screenshot from 2022-09-19 14-54-13

### CODE REVIEW

**Comment by nhat15** [comment]
ready

File: app/views/admin/genres/_search.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<%= search_form_for [:admin, @search], class: "input-group", method: "get" do |f| %>
+  <div class = "col-12 col-sm-12 col-md-3 col-lg-3 col-xl-3 col-xxl-3">
+    <%= f.label :name_cont, t(".search_by_name") %>
+    <%= f.text_field :name_cont, 
+    placeholder: "Search by name", class: "form-control title-input" %>
```
Comment (huongnt-2545) [inline app/views/admin/genres/_search.html.erb]: sai indent + I18n

File: app/views/admin/genres/_search.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  </div>
+  <div class="col-12 col-sm-12 col-md-3 col-lg-3 col-xl-3 col-xxl-3">
+    <label class = "blank">.</label>
+    <br/>
+    <%= f.submit "Search", class: "btn btn-primary" %>
```
Comment (huongnt-2545) [inline app/views/admin/genres/_search.html.erb]: I18n

File: app/views/admin/movies/_search.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<%= search_form_for [:admin, @search], class: "input-group", method: "get" do |f| %>
+  <div class = "col-12 col-sm-12 col-md-3 col-lg-3 col-xl-3 col-xxl-3">
+    <%= f.label :title_or_description_cont, t(".search_by_name_or_descption") %>
+    <%= f.text_field :title_or_description_cont, 
+    placeholder: "Search by name or descripion", class: "form-control title-input" %>
```
Comment (huongnt-2545) [inline app/views/admin/movies/_search.html.erb]: I18n

File: db/seeds.rb [hunk 4]
```diff
@@ -1,94 +1,92 @@
-Genre.create!([{name: 'Horror'}, {name: 'Comedy'}, {name: 'Action'}, {name: 'Adventure'}, {name: 'Fantasy'}])
+# Genre.create!([{name: 'Horror'}, {name: 'Comedy'}, {name: 'Action'}, {name: 'Adventure'}, {name: 'Fantasy'}])
```
Comment (huongnt-2545) [inline db/seeds.rb]: chạy seed xong rồi thì bỏ comment đi em

**Comment by nhat15** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**